### DISCUSSION TITLE
Command-line options vs INI configuration. What are the best practices ?

### DISCUSSION THREAD

**Discussion by harmin-parra** [pytest-dev/pytest-html#717] [Q&A]
Pytest allows external parameters via

* command-line options
  `parser.addoption()  `

* configuration INI file `pytest.ini`
  `parser.addini()`

Is there a general guideline on when to use each of these 2 options ?
In which cases should I use command-line options and in which cases should I use the configuration INI file?

**Comment by BeyondEvil** [comment]
You probably want to move this discussion here: https://github.com/pytest-dev/pytest/discussions
